Web Inspector: REGRESSION (Bug 241817 ) Filter bars missing from Styles and Computed panels

rcaliman-apple opened the pull request:
#### 27afcd755691b9e231a72d7d57720827c718029c
<pre>
Web Inspector: REGRESSION (Bug 241817 ) Filter bars missing from Styles and Computed panels
<a href="https://bugs.webkit.org/show_bug.cgi?id=242186">https://bugs.webkit.org/show_bug.cgi?id=242186</a>

Reviewed by Devin Rousso and Patrick Angle.

The visibility of the filter bar was guarded on the truthiness of `this._filterBar`,
a check that occured in `_showPanel()` before the filter bar was created in `initialLayout()`.
This resulted in a filter bar that was eventually created but remained hidden.

This patch inlines the contents of `WI.GeneralStyleDetailsSidebarPanel._showPanel()`
into `WI.GeneralStyleDetailsSidebarPanel.initialLayout()` ensuring the filter bar is
always visible if created.

* Source/WebInspectorUI/UserInterface/Views/GeneralStyleDetailsSidebarPanel.js:
 (WI.GeneralStyleDetailsSidebarPanel.prototype.initialLayout):
 (WI.GeneralStyleDetailsSidebarPanel.prototype._showPanel): Deleted.

Canonical link: <a href="https://commits.webkit.org/252093@main">https://commits.webkit.org/252093@main</a>
</pre>

dcrousso reviewed (approved):
i wonder if instead we should just switch back to checking `!!this._panel.filterDidChange`.  i do like the existing approach tho in that it only has to check that once 🤔

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/GeneralStyleDetailsSidebarPanel.js:
```diff
@@ -128,8 +128,6 @@ WI.GeneralStyleDetailsSidebarPanel = class GeneralStyleDetailsSidebarPanel exten
 
     initialLayout()
     {
-        this._showPanel(this._panel);
```
oh wait actually i think this might cause issues cause now `this._panel` will be in a different spot in the DOM (i.e. last instead of first)

i think we could/should just remove `_showPanel` entirely by inlining `this.contentView.addSubview(this._panel)` here and `this.contentView.element.classList.add("has-filter-bar");` right after `this._filterBar` is created

rcaliman-apple replied:
That sounds good. 

I inlined the relevant contents of `_showPanel()` into `initialLayout()`, preserving DOM order.

I dropped the toggling of `WI.GeneralStyleDetailsSidebarPanel.FilterInProgressClassName` because there's no way a newly created filter bar can have active filters.

rcaliman-apple committed:
Web Inspector: REGRESSION (Bug 241817 ) Filter bars missing from Styles and Computed panels
https://bugs.webkit.org/show_bug.cgi?id=242186

Reviewed by Devin Rousso and Patrick Angle.

The visibility of the filter bar was guarded on the truthiness of `this._filterBar`,
a check that occured in `_showPanel()` before the filter bar was created in `initialLayout()`.
This resulted in a filter bar that was eventually created but remained hidden.

This patch inlines the contents of `WI.GeneralStyleDetailsSidebarPanel._showPanel()`
into `WI.GeneralStyleDetailsSidebarPanel.initialLayout()` ensuring the filter bar is
always visible if created.

* Source/WebInspectorUI/UserInterface/Views/GeneralStyleDetailsSidebarPanel.js:
 (WI.GeneralStyleDetailsSidebarPanel.prototype.initialLayout):
 (WI.GeneralStyleDetailsSidebarPanel.prototype._showPanel): Deleted.

Canonical link: https://commits.webkit.org/252093@main
